pycryptobot.py

```diff
@@ -56,35 +56,66 @@ def executeJob(sc=None, app: PyCryptoBot=None, state: AppState=None, trading_dat
 
     # analyse the market data
     if app.isSimulation() and len(trading_data.columns) > 8:    
+													
         df = trading_data
+														  
+																				 
         # if smartswitch then get the market data using new granularity
         if app.sim_smartswitch:
-            df_last = app.getInterval(df, state.iterations - 1)
+            df_last = app.getInterval(df, state.iterations)
             if len(df_last.index.format()) > 0:
                 if app.simstartdate is not None:
-                    startDate = app.simstartdate
+                    startDate = app.simstartdate.replace("T", " ")
```

Comment:
same as above, please, don't do that replacement all over the code base, other people might not understand when this replacement is needed and when not, and make mistakes.